### PR TITLE
Custom apps ZIM migration

### PR DESCRIPTION
**Pull request by BPerlakiH**
Fix for #601 - Migrating bookmarks for custom app.

There were a couple of thing to do:
- make sure we migrate the bookmarks
- migrate the tabs, which have "interactive data" - a kind of snapshot of the tab for webView, in a form of Data, which is now migrating at the point of opening the URL from that data
- make sure we delete the old file reference in the DB - as currently we were displaying a list of local ZIM files in the menu, (up to 5). For custom apps it should really be only 1 item, so we make sure we remove the reference to the old ZIM file.

### CODE REVIEW

**Comment by kelson42** [comment]
@BPerlakiH Can you please fix the codefactor?

**Comment by kelson42** [comment]
@BPerlakiH Ready to review?

**Comment by BPerlakiH** [comment]
There's one more thing need to be fixed here, which is the "interactive data" saved together with the state of each tab.
Will update the PR soon.

**Comment by BPerlakiH** [comment]
@kelson42 @rgaudin it is ready for review.

File: SwiftUI/Model/ZimMigration.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                  let latest = zimFiles.popLast() else {
+                return
+            }
+
+            context.mergePolicy = NSMergePolicy.mergeByPropertyObjectTrump
+            for zimFile in zimFiles {
+                migrateFrom(zimFile: zimFile, toZimFile: latest, using: context)
+            }
+        }
+        debugPrint("migrationForCustomApps Complete")
```
Comment (rgaudin) [inline SwiftUI/Model/ZimMigration.swift:35]: Is this supposed to stay?

**Review by rgaudin — APPROVED** [review]
Can't say much

File: SwiftUI/Model/ZimMigration.swift (hunk 1 above)
Comment (BPerlakiH) [reply SwiftUI/Model/ZimMigration.swift]: Thanks, good catch, I've removed it now.

**Review by rgaudin — APPROVED** [review]

**Pull request merged**